src/Team/team.controller.ts

```diff
@@ -67,60 +80,45 @@ export const deleteTeam = async (req: Request, res: Response) => {
 };
 //update
 export const addPlayer = async (req: Request, res: Response) => {
-  console.log(req.body);
   const teamId: string = req.params.teamId;
   const teamInt: number = parseInt(teamId);
-  const playersNumber: number[] = await getPlayersNumberInTeamManger(teamInt);
-  const playerId =req.body.playerlist;
-  const playerNumber = await getCurrentPlayerNumberInManager(playerId);
-  let myid:Iplayer= req.body.playerlist
+  const myId: number[][] = await getIds(teamInt);
+  const objOfplayersNumber: { playersNumber: number[] }[] = await getPlayersNumberInTeamManger(teamInt);
+  const playersnumber: number[] = objOfplayersNumber[0].playersNumber;
+  const playerId = req.body.playerlist;
+  const objplayerNumber = await getCurrentPlayerNumberInManager(playerId);
+  const playerNumber: number = objplayerNumber;
+  // let checkDuplicate =(array) =>new Set(array).size !== array.lenght;
+  const currentPlayerid:number= req.body.playerlist;
+  const currentPlayer:Iplayer = await PlayerSchema.findOne({playerId:currentPlayerid});
```

Comment:
The controller should only accept the parameters and control the logic (If there is a lot of logic - we add a manager). Don't use mongoose here. only in the repository.